src/test/ui/lint/issue-47390-unused-variable-in-struct-pattern.stderr

```diff
@@ -0,0 +1,40 @@
+warning: unused variable: `i_think_continually`
+  --> $DIR/issue-47390-unused-variable-in-struct-pattern.rs:22:9
+   |
+22 |     let i_think_continually = 2;
+   |         ^^^^^^^^^^^^^^^^^^^ help: consider using `_i_think_continually` instead
+   |
+note: lint level defined here
+  --> $DIR/issue-47390-unused-variable-in-struct-pattern.rs:13:9
+   |
+13 | #![warn(unused)] // UI tests pass `-A unused` (#43896)
+   |         ^^^^^^
+   = note: #[warn(unused_variables)] implied by #[warn(unused)]
+
+warning: unused variable: `corridors_of_light`
+  --> $DIR/issue-47390-unused-variable-in-struct-pattern.rs:29:26
+   |
+29 |     if let SoulHistory { corridors_of_light,
+   |                          ^^^^^^^^^^^^^^^^^^ help: try ignoring the field: `corridors_of_light: _`
+
+warning: variable `hours_are_suns` is assigned to, but never used
+  --> $DIR/issue-47390-unused-variable-in-struct-pattern.rs:30:26
+   |
+30 |                          mut hours_are_suns,
+   |                          ^^^^^^^^^^^^^^^^^^
+   |
+   = note: consider using `_hours_are_suns` instead
```

Comment:
This warning is emitted somewhere else. Would you mind checking how hard it'd be to modify that note by checking if it is a field in a struct literal?

Reply:
@estebank The PR [touches this code](https://github.com/rust-lang/rust/pull/47922/files#diff-4127ea26dee4e305dffd37e9512dcbb7R1515) (on master, the note starts with "to avoid this warning", which I thought was unnecessary). I had _started_ working on making this a structured suggestion, too, but then backed off when I realized that getting the suggestion right would involve more refactoring than I wanted to do now. Notice that the span includes the `mut`: in the case of an entirely unused variable, that's fine (the mut is also unused, so replacing both the variable and any `mut`s or `ref`s with `variable: _` is correct), but in the case of a mere unused _assignment_ (where we mutate the variable, but don't do anything interesting with it), not only is the `mut` significant, but there are also other appearances of the variable (requiring a multi-span suggestion).